src/SetupApi/SetupApi+SP_DRVINFO_DETAIL_DATA.cs

```diff
@@ -0,0 +1,167 @@
+            /// If the CompatIDs list is not empty (that is, if <see cref="CompatIDsLength"/> is not zero), the CompatIDs list starts at offset
+            /// <see cref="CompatIDsOffset"/> from the beginning of this buffer, and is terminated with an additional <see langword="null"/>
+            /// character at the end of the list.
+            /// </remarks>
+            public fixed char HardwareID[4];
+
+            /// <summary>
+            /// Gets a <see cref="string"/> that contains the name of the INF DDInstall section for this driver.
+            /// This must be the basic DDInstall section name, such as InstallSec, without any OS/architecture-specific extensions.
+            /// </summary>
+            public string SectionNameString
+            {
+                get
+                {
+                    fixed (char* sectionName = this.SectionName)
+                    {
+                        return new string(sectionName);
+                    }
+                }
+            }
+
+            /// <summary>
+            /// Gets a <see cref="string"/> that contains the full-qualified name of the INF file for this driver.
+            /// </summary>
+            public string InfFileNameString
+            {
+                get
+                {
+                    fixed (char* infFileName = this.InfFileName)
+                    {
+                        return new string(infFileName);
+                    }
+                }
+            }
+
+            /// <summary>
+            /// Gets a <see cref="string"/> that describes the driver.
+            /// </summary>
+            public string DrvDescriptionString
+            {
+                get
+                {
+                    fixed (char* desription = this.DrvDescription)
+                    {
+                        return new string(desription);
+                    }
+                }
+            }
+
+            /// <summary>
+            /// Gets the hardware ID in the INF Models section.
+            /// </summary>
+            public string HardwareIDString
+            {
+                get
+                {
+                    fixed (char* hardwareId = this.HardwareID)
+                    {
+                        return new string(hardwareId);
+                    }
+                }
+            }
+
+            /// <summary>
+            /// Initializes a new instance of the <see cref="SP_DRVINFO_DETAIL_DATA" /> struct
+            /// with <see cref="SP_DRVINFO_DETAIL_DATA.cbSize" /> set to the correct value.
+            /// </summary>
+            /// <returns>An instance of <see cref="SP_DRVINFO_DETAIL_DATA"/>.</returns>
+            public static SP_DRVINFO_DETAIL_DATA Create()
+            {
+                var size = Marshal.SizeOf(typeof(SP_DRVINFO_DETAIL_DATA));
+
+                if (!Environment.Is64BitProcess)
+                {
+                    // On 32-bit Windows
+                    size -= 6;
```

Comment:
IIRC, we got this struct to work "without tricks" on 32-bit Windows by setting `Pack = 1` in the `StructLayout` attribute, and `HardwareID[1]`, and `Pack = 4` and `HardwareId[4]` for 64-bit Windows.

Reply:
Ya, not a fan of these tricks, particularly when I don't understand why they're necessary since I don't know when this might change to something that breaks (e.g. Windows on ARM)?